### PR TITLE
feat: add C implementation of `stats/base/dists/rayleigh/entropy`

### LINKED ISSUE #3799: [RFC]: Add C implementation for `@stdlib/stats/base/dists/rayleigh/entropy`
### Description

This RFC proposes adding a C implementation, including a Node.js native addon, C benchmarks, and C examples, for the [`@stdlib/stats/base/dists/rayleigh/entropy`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy) package.

When adding support, the following tasks should be completed:

1. A C source implementation should be added to a `src` folder within the package's directory.
2. A native add-on interface should be added in order to call the C implementation from JavaScript.
3. A JavaScript file should be added to the `lib` folder to allow benchmarking and unit testing the native implementation from JavaScript.
4. A new test file should be created which includes unit tests testing against expected behavior.
5. C benchmarks should be added which measure performance for the C source implementation.
6. A file containing C examples should be added to demonstrate example usage.
7. JavaScript benchmarks should be added which measure performance for the native add-on.
8. Build and configuration files should be added to allow compiling and running the C source implementation using project `make` commands (as documented below).
9. The README should be updated to include documentation for the added C API.

To provide a concrete example of what a PR adding the desired method should contain, see https://github.com/stdlib-js/stdlib/pull/3354, which is a PR adding a C implementation for the CDF of an arcsine distribution. This should provide an idea of what is expected.

#### Prerequisites

1. Review JavaScript implementation in `lib/main.js` and its required modules to identify all `stdlib` functions used.
2. Verify C implementations exist for each required function.

### Related Issues

#3355

### Questions

No.

### Other

Once the implementation is ready including C examples and benchmarks, we should be able to run the following `make` commands.

#### Build native add-on

```bash
NODE_ADDONS_PATTERN="@stdlib/stats/base/dists/rayleigh/entropy" make install-node-addons
```

#### Run C examples

```bash
make examples-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/examples/c/example.c"
```

#### Run C benchmarks

```bash
make benchmark-c-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/c/benchmark.c"
```

#### Run JavaScript benchmarks

```bash
make benchmark-javascript-files FILES="$(pwd)/lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/benchmark.native.js"
```

#### Run all tests

```bash
make TESTS_FILTER=".*/@stdlib/stats/base/dists/rayleigh/entropy/.*" test
```

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please minimize the submission of LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [x] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [x] Searched for existing issues and pull requests.
- [x] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Vinit-Pandit**
Resolves #3799 .

## Description

> What is the purpose of this pull request?

This pull request:

-   add c implementation of stats/base/dists/rayleigh/entropy

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3799 
## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/README.md [hunk 1]
```diff
@@ -116,6 +116,98 @@ for ( i = 0; i < 10; i++ ) {
...
+
+int main( void ) {
+    double sigma;
+    double y;
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        sigma = random_uniform( 0.0, 20.0 );
+        y = stdlib_base_dists_rayleigh_entropy( sigma );
+        printf( "σ: %lf, h(x;σ): %lf\n", sigma, y );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/README.md]:
```suggestion
        printf( "σ: %lf, h(σ): %lf\n", sigma, y );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/examples/c/example.c [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+int main( void ) {
+	double sigma;
+	double y;
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		sigma = random_uniform( 0.0, 20.0 );
+		y = stdlib_base_dists_rayleigh_entropy( sigma );
+		printf( "σ: %lf, h(x;σ): %lf\n", sigma, y );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/examples/c/example.c]:
```suggestion
		printf( "σ: %lf, h(σ): %lf\n", sigma, y );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/src/main.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+*
+* @param x        input value
+* @return         evaluated entropy
+*
+* @example
+* double y = stdlib_base_arcsine_entropy( 9.0 );
+* // returns ~3.139
+*/
+double stdlib_base_dists_rayleigh_entropy( const double sigma ) {
+	double s2;
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/src/main.c]:
```suggestion
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/benchmark.js [hunk 4]
```diff
@@ -32,13 +33,19 @@ var entropy = require( './../lib' );
...
+	sigma = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		sigma[ i ] = ( randu()*20.0 ) + EPS;
+	}
+
 	b.tic();
 	for ( i = 0; i < b.iterations; i++ ) {
-		sigma = ( randu()*20.0 ) + EPS;
-		y = entropy( sigma );
+		y = entropy( sigma[ i % 100 ] );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/benchmark.js]:
```suggestion
		y = entropy( sigma[ i % len ] );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/benchmark.native.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+	len = 100;
+	sigma = new Float64Array( len );
+	for ( i = 0; i < len; i++ ) {
+		sigma[ i ] = ( randu() * 20.0 ) + EPS;
+	}
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		y = entropy( sigma[ i % 100] );
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/benchmark.native.js]:
```suggestion
		y = entropy( sigma[ i % len ] );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/src/main.c [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/math/base/special/ln.h"
+#include "stdlib/constants/float64/eulergamma.h"
+#include "stdlib/constants/float64/sqrt_two.h"
+
+
+/**
+* Returns the differential entropy of a Rayleigh distribution.
+*
+* @param x        input value
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/src/main.c]:
```suggestion
* @param sigma        input value
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/lib/native.js [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+// MODULES //
+
+var addon = require( './../src/addon.node' );
+
+
+// MAIN //
+
+/**
+* Returns the differential entropy of a Rayleigh distribution.
+*
```
Comment (Vinit-Pandit) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/lib/native.js:30]:
```suggestion
*
* @private
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/README.md [hunk 8]
```diff
@@ -116,6 +116,98 @@ for ( i = 0; i < 10; i++ ) {
...
+
+<section class="usage">
+
+### Usage
+
+```c
+#include "stdlib/stats/base/dists/rayleigh/entropy.h"
+```
+
+#### stdlib_base_dists_rayleigh_entropy( x, sigma )
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/README.md]:
```suggestion
#### stdlib_base_dists_rayleigh_entropy( sigma )
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/c/benchmark.c [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+static double benchmark( void ) {
+	double elapsed;
+	double sigma[ 100 ];
+	double y;
+	double t;
+	int i;
+
+
+	for ( i = 0; i < 100; i++ ) {
+		sigma[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS , STDLIB_CONSTANT_FLOAT64_EPS + 20 );
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/benchmark/c/benchmark.c]:
```suggestion
		sigma[ i ] = random_uniform( STDLIB_CONSTANT_FLOAT64_EPS , STDLIB_CONSTANT_FLOAT64_EPS + 20.0 );
```

File: lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/src/main.c [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+#include "stdlib/math/base/special/ln.h"
+#include "stdlib/constants/float64/eulergamma.h"
+#include "stdlib/constants/float64/sqrt_two.h"
+
+
+/**
+* Returns the differential entropy of a Rayleigh distribution.
+*
+* @param sigma        input value
+* @return         evaluated entropy
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/stats/base/dists/rayleigh/entropy/src/main.c]: Updating parameter description and fixing horizontal alignment:

```suggestion
* @param sigma    scale parameter
* @return         evaluated entropy
```

**Review by Planeshifter — APPROVED** [review]
Thank you @MeastroZI! Appreciate your proactive updates with improvements to the PR. It's looking good overall; I noticed a few more small things but otherwise this should be ready to land.

**Comment by Vinit-Pandit** [comment]
Thank You @Planeshifter , I will be more carefull from next time for this kind of issues.

**Pull request merged**